etcdserver/server.go

```diff
@@ -786,6 +834,20 @@ func (s *EtcdServer) Stop() {
 	<-s.done
 }
 
+// Stop stops the server gracefully, and shuts down the running goroutine.
+// Stop should be called after a Start(s), otherwise it will block forever.
+// When stopping leader, Stop transfers its leadership to one of its peers
+// before stopping the server.
+func (s *EtcdServer) Stop() {
+	s.stopServer(true)
+}
+
+// StopWithNoLeaderTransfer stops the server without automatic leader transfer.
+// It's useful for shutting down the testing cluster.
+func (s *EtcdServer) StopWithNoLeaderTransfer() {
+	s.stopServer(false)
```

Comment:
actually, why does there have to be a separate `Stop()` function? can't `Stop()` keep its old behavior and instead have

```go
s.TransferLeadership(...)
s.Stop()
```

if a transfer is needed before a stop?